src/neural/network.h

```diff
@@ -107,6 +107,7 @@ class Network {
   virtual const NetworkCapabilities& GetCapabilities() const = 0;
   virtual std::unique_ptr<NetworkComputation> NewComputation() = 0;
   virtual void InitThread(int /*id*/) {}
+  virtual int GetMiniBatchSize() const { return 256; }
```

Comment:
I wonder whether it would be cleaner to add the recommended minibatch size into the `NetworkCapabilities` rather than adding a new function (not sure which variant I prefer though).

Reply:
The 'NetworkCapabilities' so far only stores properties of the weights file, and I have plans to expand the interface to pass the search minibatch-size value so that the backend can adjust the memory buffer allocations as needed.

Reply:
Maybe `NetworkCapabilities` is a poor name, but it still looks cleaner to have all information that backend tells to the calling code in a structure, to make interface less leaky.
It's fine to do refactor this later though.

Reply:
I'm warming up to the idea, but I want to finish with the rest of my plan before doing it. There are some cornet cases I haven't though through yet.